rust/lancedb/src/remote/client.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024 LanceDB Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+use reqwest::{
+    header::{HeaderMap, HeaderValue},
+    RequestBuilder, Response,
+};
+
+use crate::error::{Error, Result};
+
+#[derive(Debug)]
+pub struct RestfulLanceDbClient {
+    client: reqwest::Client,
+    host: String,
+}
+
+impl RestfulLanceDbClient {
+    fn default_headers(
+        api_key: &str,
+        region: &str,
+        db_name: &str,
+        has_host_override: bool,
```

Comment:
If the user has overridden the host then we set the `x-lancedb-database` header.  I copied this from the existing python client.  I believe the rationale is that we normally get the db header from the URI.  However, when testing locally, we don't have DNS setup and we need to fallback to some alternate method.

It would probably be simpler, and I think it would be safe, if we just always set this header too.  CC @albertlockett @chebbyChefNEQ for second opinion.